webapp/src/extensionsBrowser.tsx

```diff
@@ -408,71 +405,148 @@ export const ExtensionsBrowser = (props: ExtensionsProps) => {
-                </div>
-                {currentTab == TabState.Recommended && preferredExts.map(e => <ExtensionCard scr={e} name={e.name} onCardClick={installExtension} imageUrl={e.imageUrl} description={e.description}
-                        learnMoreUrl={e.fullName ? `/pkg/${e.fullName}`: undefined} loading={e.loading} />)}
-                {currentTab == TabState.Installed && installedExtensions.map(e =>
-                        <ExtensionCard scr={e} name={e.name} onCardClick={() => handleInstalledCardClick(e)} imageUrl={e.imageUrl} description={e.description} learnMoreUrl={e.fullName ? `/pkg/${e.fullName}` : undefined}/>)}
-                {currentTab == TabState.InDevelopment && local.forEach(p => {
-                            <ExtensionCard scr={p} name={p.name} description={lf("Local copy of {0} hosted on github.com", p.githubId)} onCardClick={addLocal} />
-                        })}
+                {displayMode == ExtensionView.Search &&
+                    <div className="extension-display">
+                        <div className="breadcrumbs">
+                            <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
+                        </div>
+                        <div className="ui cards left">
+                            {extensionsToShow?.map(scr =>
+                                <ExtensionCard
+                                    key={scr.loading ? `loading ${loadingCardId++}`: 'searched:' + scr.name}
+                                    name={scr.name ?? `loading ${loadingCardId++}`}
+                                    description={scr.description}
+                                    imageUrl={scr.imageUrl}
+                                    scr={scr}
+                                    onCardClick={installExtension}
+                                    learnMoreUrl={scr.fullName ? `/pkg/${scr.fullName}` : undefined}
+                                    loading={scr.loading}
+                                    role="button"
+                                />)}
+                        </div>
+                    </div>}
+                {displayMode == ExtensionView.Tags &&
+                    <div className="extension-display">
+                        <div className="breadcrumbs">
+                            <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
+                            <span>/</span>
+                            <span>{selectedTag}</span>
+                        </div>
+                        <div className="ui cards left">
+                            {extensionsToShow?.map(scr =>
+                                <ExtensionCard
+                                    key={scr.loading ? `loading ${loadingCardId++}`: 'tagged:' + scr.name}
```

Comment:
you can probably just pass the index here. i'm not sure you need the loading/tagged bit either